rasa/nlu/classifiers/sklearn_intent_classifier.py

```diff
@@ -225,40 +244,54 @@ def predict(self, X: np.ndarray) -> Tuple[np.ndarray, np.ndarray]:
         sorted_indices = np.fliplr(np.argsort(pred_result, axis=1))
         return sorted_indices, pred_result[:, sorted_indices]
 
-    def persist(self, file_name: Text, model_dir: Text) -> Optional[Dict[Text, Any]]:
+    def persist(self) -> None:
         """Persist this model into the passed directory."""
+        with self._model_storage.write_to(self._resource) as model_dir:
+            file_name = self.__class__.__name__
+            classifier_file_name = model_dir / f"{file_name}_classifier.pkl"
+            encoder_file_name = model_dir / f"{file_name}_encoder.pkl"
 
-        classifier_file_name = file_name + "_classifier.pkl"
-        encoder_file_name = file_name + "_encoder.pkl"
-        if self.clf and self.le:
-            io_utils.json_pickle(
-                os.path.join(model_dir, encoder_file_name), self.le.classes_
-            )
-            io_utils.json_pickle(
-                os.path.join(model_dir, classifier_file_name), self.clf.best_estimator_
-            )
-        return {"classifier": classifier_file_name, "encoder": encoder_file_name}
+            if self.clf and self.le:
+                io_utils.json_pickle(encoder_file_name, self.le.classes_)
+                io_utils.json_pickle(classifier_file_name, self.clf.best_estimator_)
 
     @classmethod
     def load(
         cls,
-        meta: Dict[Text, Any],
-        model_dir: Text,
-        model_metadata: Optional[Metadata] = None,
-        cached_component: Optional["SklearnIntentClassifier"] = None,
+        config: Dict[Text, Any],
+        model_storage: ModelStorage,
+        resource: Resource,
+        execution_context: ExecutionContext,
         **kwargs: Any,
-    ) -> "SklearnIntentClassifier":
+    ) -> SklearnIntentClassifierGraphComponent:
         """Loads trained component (see parent class for full docstring)."""
         from sklearn.preprocessing import LabelEncoder
 
-        classifier_file = os.path.join(model_dir, meta["classifier"])
-        encoder_file = os.path.join(model_dir, meta["encoder"])
-
-        if os.path.exists(classifier_file):
-            classifier = io_utils.json_unpickle(classifier_file)
-            classes = io_utils.json_unpickle(encoder_file)
-            encoder = LabelEncoder()
-            encoder.classes_ = classes
-            return cls(meta, classifier, encoder)
-        else:
-            return cls(meta)
+        try:
+            with model_storage.read_from(resource) as model_dir:
+                file_name = cls.__name__
+                classifier_file = list(model_dir.glob(f"**/{file_name}_classifier.pkl"))
+
+                if classifier_file:
+                    assert len(classifier_file) == 1
```

Comment:
let's not have an `assert` in production code. How about raising an exception which is then caught by the block in 292?

Reply:
Sure, what would be the exception? 
Here my intention was to make sure that `len` is not greater than 1. Perhaps I can raise a debug warning that more than 1 file were found? In theory it shouldn't be more than 1 file, maybe I'm overcomplicating matters.

Reply:
how about just throwing a `ValueError`? Your `except ValueError` would then catch it.

Reply:
Thinking more about this approach - it would return a new component instance when actually files were found that could be used to load the component. Maybe I'm imagining a scenario that actually would never happen (e.g. have more than 1 file found) 🤔 I will just remove the assertion, WDYT?

Reply:
were not testing this anywhere else - we're always relying on the assumption that only persist() writes to that directory - so I think it's ok to remove the check for "== 1" :)  ... we could also create some constants for classifier_file_name and encoder_file_name and just use those here and in persist (because we also rely on the model storage/resource to be a unique location for this component evyerwhere)